compiler-cli/src/http.rs

```diff
@@ -42,3 +40,29 @@ impl gleam_core::io::HttpClient for HttpClient {
             .map_err(Error::http)
     }
 }
+
+fn init_client() -> Result<&'static Client, Error> {
+    if let Some(client) = REQWEST_CLIENT.get() {
+        return Ok(client);
+    }
+
+    let certificate_path = std::env::var("GLEAM_CACERTS_PATH")
+        .map_err(|_| Error::CannotReadCertificate { path: "".into() })?;
+
+    let certificate_bytes =
+        std::fs::read(&certificate_path).map_err(|_| Error::CannotReadCertificate {
+            path: certificate_path.clone(),
+        })?;
+
+    let certificate =
+        Certificate::from_pem(&certificate_bytes).map_err(|_| Error::CannotReadCertificate {
+            path: certificate_path.clone(),
+        })?;
+
+    match Client::builder().add_root_certificate(certificate).build() {
+        Ok(client) => Ok(REQWEST_CLIENT.get_or_init(|| client)),
+        Err(_) => Err(Error::CannotReadCertificate {
```

Comment:
This error says the problem is with reading the certificate, but the error is actually for building the HTTP client.

If you think this error should never happen (I see the cert is checked above) then `.expect` can be used